core/src/main/java/org/elasticsearch/index/seqno/LocalCheckpointService.java

```diff
@@ -35,10 +36,8 @@
      * we keep a bit for each seq No that is still pending. to optimize allocation, we do so in multiple arrays
      * allocating them on demand and cleaning up while completed. This setting controls the size of the arrays
      */
-    public static String SETTINGS_BIT_ARRAYS_SIZE = "index.seq_no.checkpoint.bit_arrays_size";
-
-    /** default value for {@link #SETTINGS_BIT_ARRAYS_SIZE} */
-    final static int DEFAULT_BIT_ARRAYS_SIZE = 1024;
+    public static Setting<Integer> SETTINGS_BIT_ARRAYS_SIZE = Setting.intSetting("index.seq_no.checkpoint.bit_arrays_size", 1024,
+        4, Setting.Property.IndexScope);
```

Comment:
I reduce it in testing to something small so I can more easily create multi chunk situations. It's not really based on efficiency (everything so low is not a good idea for production anyway). Also note that these are bits, not bytes.